spotinst/resource_spotinst_ocean_spark_virtual_node_group.go

```diff
@@ -0,0 +1,181 @@
+package spotinst
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/spotinst/spotinst-sdk-go/service/ocean/spark"
+	"github.com/spotinst/spotinst-sdk-go/spotinst/client"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/spotinst/spotinst-sdk-go/spotinst"
+	"github.com/spotinst/terraform-provider-spotinst/spotinst/commons"
+	"github.com/spotinst/terraform-provider-spotinst/spotinst/ocean_spark_virtual_node_group"
+)
+
+func resourceSpotinstOceanSparkVirtualNodeGroup() *schema.Resource {
+	setupOceanSparkVirtualNodeGroupResource()
+
+	return &schema.Resource{
+		CreateContext: resourceSpotinstSparkVirtualNodeGroupCreate,
```

Comment:
should we call it 'resourceSpotinstSparkClusterVirtualNodeGroupCreate' to be consistent?